File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> r;
    vector<int> t;
    for(int i=0;i<commands.size();i++) {
        t={};
        for(int j=commands[i][0]-1;j<commands[i][1];j++) {
            t.push_back(array[j]);
        }
        sort(t.begin(),t.end(),less<int>());
        r.push_back(t[commands[i][2]-1]);
    }
    return r;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.Arrays;
import java.util.Collections;
class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] rlt = new int[commands.length];
        int[] slicedArray={};
        for(int i=0;i<commands.length;i++) {
            slicedArray=Arrays.copyOfRange(array,commands[i][0]-1,commands[i][1]);
            Arrays.sort(slicedArray);
            rlt[i]=slicedArray[commands[i][2]-1];
        }
        return rlt;
    }
}